langlib/lang.query/src/main/ballerina/helpers.bal

```diff
@@ -83,7 +84,15 @@ function getStreamFromPipeline(_StreamPipeline pipeline) returns stream<Type, Co
     return pipeline.getStream();
 }
 
-function toArray(stream<Type, CompletionType> strm, Type[] arr) returns Type[]|error {
+function toArray(stream<Type, CompletionType> strm, Type[] arr, boolean isReadOnly) returns Type[]|error {
+    if isReadOnly {
+        Type[] arr2 = [];
```

Comment:
Why do we need to create a new array/map/table type here? can't we use the structure we pass via the query desugar?

Reply:
Since it is an immutable array, we can't update it.